src/compiler/scanner.ts

```diff
@@ -998,6 +1033,25 @@ module ts {
                     return token = SyntaxKind.EndOfFileToken;
                 }
                 let ch = text.charCodeAt(pos);
+
+                if (inJSXChild) {
+                    switch (ch) {
+                        case CharacterCodes.closeBrace:
+                            return pos++, token = SyntaxKind.CloseBraceToken;
+                        case CharacterCodes.openBrace:
+                            return pos++, token = SyntaxKind.OpenBraceToken;
+                        case CharacterCodes.lessThan:
+                            if (text.charCodeAt(pos + 1) === CharacterCodes.slash) {

```

Comment:
in case of something like : 
```
<foo>test </foo/.Something()
```
We could try something with closing tag parsing, for example if we detect an `/` instead of the last `>`  in the closing element parseJSXElement could return `null` when speculative ?